src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -877,3 +877,48 @@ def _validator(namespace):
 
         setattr(namespace, property_name, service_types)
     return _validator
+
+
+def page_blob_tier_validator(namespace):
+    if not namespace.tier:
+        return
+
+    if namespace.blob_type != 'page' and namespace.tier:
+        raise ValueError('Blob tier is only applicable to page blobs on premium storage accounts.')
+
+    if namespace.tier.startswith('_'):
+        raise ValueError('Unknown premium page blob tier name.')
```

Comment:
It's not enum but a class with string as properties. So i didn't use the enum list.